File: 7-kyu/80's-kids-#1--how-many-licks-does-it-take-/javascript/solution.js
const totalLicks = env => {
  const avgLicksBase = 252;
  const totalEnvLicks = Object.values(env).reduce((acc, nxt) => acc + nxt, 0) || 0;
  let additionalChallenge = "";
  if (Object.values(env).some(licks => licks > 0)) {
    let challengeType = Object.keys(env)[0];
    let mostLicks = Object.values(env)[0];
    for (const [challenge, licks] of Object.entries(env)) {
      if (licks > mostLicks) {
        challengeType = challenge;
        mostLicks = licks;
      }
    }
    additionalChallenge = ` The toughest challenge was ${challengeType}.`;
  }
  return `It took ${avgLicksBase + totalEnvLicks} licks to get to the tootsie roll center of a tootsie pop.` + additionalChallenge;
};

File: 7-kyu/80's-kids-#1--how-many-licks-does-it-take-/python/solution.py
def total_licks(env):
    licks_base = 252
    total_licks = sum(env.values())
    additional_challenge = ''
    if any(l > 0 for l in env.values()):
        chal_type = list(env.keys())[0]
        greatest_licks = list(env.values())[0]
        for chal, licks in env.items():
            if licks > greatest_licks:
                chal_type = chal
                greatest_licks = licks
        additional_challenge = f' The toughest challenge was {chal_type}.'
    return f'It took {licks_base + total_licks} licks to get to the tootsie roll center of a tootsie pop.{additional_challenge}'
